frontend/src/services/ApiService.ts

```
@@ -78,3 +78,34 @@ ApiService.interceptors.response.use(
     throw transformedError
   },
 )
+
+export const processFetchResponse = async (response: Response) => {
+  try {
+    const data = await response.json()
+
+    // TODO: data may be present but if response is non 2XX, transform to error and throw
+    return data
+  } catch (error) {
+    if (error instanceof Error) {
+      datadogLogs.logger.warn(`Fetch error: ${error.message}`, {
+        meta: {
+          action: 'processFetchResponse',
+          response: {
+            status: response.status,
+            statusText: response.statusText,
+            headers: [...(response.headers?.entries() || [])],
+            body: await response.text(),
+          },
+          error: {
+            name: error.name,
+            message: error.message,
+            stack: error.stack,
+            dump: JSON.stringify(error),
+          },
+        },
+      })
+
+      throw error
```

This `throw` should be out of the `if`, otherwise the code is swallowing the error, and will likely cause runtime errors down the processing chain when the code assumes it was a response when it does not.

---

[541bf25](https://github.com/opengovsg/FormSG/pull/6008/commits/541bf25cefe0bd9526f7197b32409e7958124cf8)